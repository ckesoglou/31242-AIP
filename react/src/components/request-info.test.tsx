import React from "react";
import { shallow } from "enzyme";
import RequestInfo from "./request-info";

describe("<RequestInfo />", () => {
  it("should render correctly", () => {
    const wrapper = shallow(
      <RequestInfo
        request={{
          id: "1",
          author: { username: "James", display_name: "James" },
          completed_by: { username: "Kevin", display_name: "Kevin" },
          proof_of_completion: "Some ID",
          rewards: [
            { id: "1", display_name: "Hug" },
            { id: "2", display_name: "Coffee" },
          ],
          details: "Clean the fridge",
          created_time: "02/02/2020",
          comletion_time: "02/02/2020",
          is_completed: true,
        }}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
