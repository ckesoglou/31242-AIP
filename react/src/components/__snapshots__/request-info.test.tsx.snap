// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RequestInfo /> should render correctly 1`] = `
<div
  id="requestItem"
>
  <InfoIcon
    className="cursorPointer"
    color="primary"
    fontSize="large"
    onClick={[Function]}
  />
  <WithStyles(ForwardRef(Dialog))
    TransitionComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "muiSupportAuto": true,
        "propTypes": Object {
          "children": [Function],
          "disableStrictModeCompat": [Function],
          "in": [Function],
          "onEnter": [Function],
          "onEntered": [Function],
          "onEntering": [Function],
          "onExit": [Function],
          "onExited": [Function],
          "onExiting": [Function],
          "style": [Function],
          "timeout": [Function],
        },
        "render": [Function],
      }
    }
    fullWidth={true}
    id="completeForm"
    maxWidth="xs"
    onClose={[Function]}
    open={false}
    scroll="body"
  >
    <WithStyles(ForwardRef(DialogTitle))
      disableTypography={true}
    >
      <WithStyles(ForwardRef(Typography))
        id="modalTitle"
        variant="h5"
      >
        Request Details
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(DialogTitle))>
    <WithStyles(ForwardRef(Divider)) />
    <WithStyles(ForwardRef(DialogContent))
      className="content"
      id="requestDetailsTable"
    >
      <WithStyles(ForwardRef(TableContainer))
        id="infoTable"
      >
        <WithStyles(ForwardRef(Table))>
          <WithStyles(ForwardRef(TableBody))>
            <WithStyles(ForwardRef(TableRow))>
              <WithStyles(ForwardRef(TableCell))
                align="center"
                className="infoTableCell"
                id="infoTableTitle"
              >
                Created by:
              </WithStyles(ForwardRef(TableCell))>
              <WithStyles(ForwardRef(TableCell))
                align="center"
                className="infoTableCell"
                id="infoTableContent"
              >
                James
              </WithStyles(ForwardRef(TableCell))>
            </WithStyles(ForwardRef(TableRow))>
            <WithStyles(ForwardRef(TableRow))>
              <WithStyles(ForwardRef(TableCell))
                align="center"
                id="infoTableTitle"
              >
                Task:
              </WithStyles(ForwardRef(TableCell))>
              <WithStyles(ForwardRef(TableCell))
                align="center"
                id="infoTableContent"
              >
                Clean the fridge
              </WithStyles(ForwardRef(TableCell))>
            </WithStyles(ForwardRef(TableRow))>
            <WithStyles(ForwardRef(TableRow))>
              <WithStyles(ForwardRef(TableCell))
                align="center"
                id="infoTableTitle"
              >
                Created on:
              </WithStyles(ForwardRef(TableCell))>
              <WithStyles(ForwardRef(TableCell))
                align="center"
                id="infoTableContent"
              >
                02/02/2020
              </WithStyles(ForwardRef(TableCell))>
            </WithStyles(ForwardRef(TableRow))>
            <WithStyles(ForwardRef(TableRow))>
              <WithStyles(ForwardRef(TableCell))
                align="center"
                id="infoTableTitle"
              >
                Completed:
              </WithStyles(ForwardRef(TableCell))>
              <WithStyles(ForwardRef(TableCell))
                align="center"
                id="infoTableContent"
              >
                Yes
              </WithStyles(ForwardRef(TableCell))>
            </WithStyles(ForwardRef(TableRow))>
          </WithStyles(ForwardRef(TableBody))>
        </WithStyles(ForwardRef(Table))>
      </WithStyles(ForwardRef(TableContainer))>
    </WithStyles(ForwardRef(DialogContent))>
    <WithStyles(ForwardRef(DialogContent))
      id="requestInfoRewards"
    >
      <WithStyles(ForwardRef(DialogContentText))
        variant="h6"
      >
        Rewards
      </WithStyles(ForwardRef(DialogContentText))>
      <IouFavourSingle
        giverDisplayName="James Lee"
        item={
          Object {
            "display_name": "Hug",
            "id": "1",
          }
        }
        key="1"
        recieverDisplayName="Kevin"
      />
      <IouFavourSingle
        giverDisplayName="James Lee"
        item={
          Object {
            "display_name": "Coffee",
            "id": "2",
          }
        }
        key="2"
        recieverDisplayName="Kevin"
      />
    </WithStyles(ForwardRef(DialogContent))>
    <WithStyles(ForwardRef(Divider)) />
    <WithStyles(ForwardRef(DialogActions))>
      <WithStyles(ForwardRef(Button))
        color="primary"
        onClick={[Function]}
        size="large"
      >
        Thank you
      </WithStyles(ForwardRef(Button))>
    </WithStyles(ForwardRef(DialogActions))>
  </WithStyles(ForwardRef(Dialog))>
</div>
`;
